cuda_pathfinder/cuda/pathfinder/_utils/conda_env.py

```diff
@@ -0,0 +1,41 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import functools
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Literal, Optional
+
+# https://docs.conda.io/projects/conda-build/en/stable/user-guide/environment-variables.html
+
+BUILD_STATES = ("RENDER", "BUILD", "TEST")
+
+
+@dataclass(frozen=True)
+class CondaPrefix:
+    env_state: Literal["RENDER", "BUILD", "TEST", "activated"]
+    path: Path
+
+
+@functools.cache
+def get_conda_prefix() -> Optional[CondaPrefix]:
+    """
+    Return the effective conda prefix.
+    - RENDER, BUILD, TEST: inside conda-build (host prefix at $PREFIX)
+    - activated: user-activated env ($CONDA_PREFIX)
+    - None: neither detected
+    """
+    state = os.getenv("CONDA_BUILD_STATE")
+    if state:
+        if state in BUILD_STATES:
+            p = os.getenv("PREFIX")
```

Comment:
I think this is a bit nerve-wrecking. 

Recall that there was a term "splayed layout" from a past discussion. At the build stage, there are two environments: build and host. `$PREFIX` is for the later. But the build (compiler) toolchain, which is executed on the native platform, has headers living in `$BUILD_PREFIX`. IMHO it is where we should search first, and if not found, search `$PREFIX`.

At the test stage, there's only `$PREFIX` so the code should work fine.

FWIW, I honestly think it is far overstretching that we also support AOT use cases (ex: conda-build). It was not never my intention. I only wanted JIT compilation to work 😅  If we're building an MVP I'd rather we purge this. But if any of you have strong opinions, I won't insist. Just don't document this behavior for now.